homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,276 @@
+"""Support for EnOcean roller shutters."""
+from __future__ import annotations
+
+import asyncio
+from enum import Enum
+import logging
+
+from enocean.protocol.constants import RORG
+from enocean.protocol.packet import RadioPacket
+from enocean.utils import combine_hex
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASSES_SCHEMA,
+    PLATFORM_SCHEMA,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import SIGNAL_SEND_MESSAGE
+from .device import EnOceanEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "EnOcean roller shutter"
+
+CONF_SENDER_ID = "sender_id"
+
+WATCHDOG_TIMEOUT = 1
+WATCHDOG_INTERVAL = 0.2
+WATCHDOG_MAX_QUERIES = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Required(CONF_SENDER_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+    }
+)
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Reply:
Thank you, @frenck, I'll see what I can do about that. However, looking at the provided link, there is also a cross-reference to ADR-0010 (https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) stating that "Integrations that communicate with devices and/or services are only configured via the UI. [...] Existing integrations that should not have a YAML configuration, are allowed and encouraged to implement a configuration flow and remove YAML support. Changes to existing YAML configuration for these same existing integrations, will no longer be accepted.". Hence: Do I understand this correctly, that I would have to change the entire EnOcean integration to use UI-based configuration instead of platform YAML configuration?

Reply:
We've allowed this integration to keep the YAML settings for the existing platforms when the config flow was introduced. The plan was to continue working on moving the device configuration away from the platforms.

See https://github.com/home-assistant/core/pull/35927#discussion_r436042863

We didn't see further PRs for that work though.

Reply:
Thanks, @MartinHjelmare, for the clarification. I'll have a look into the work that needs to be done to use option flow instead of configuration.yaml.